hexa/datasets/models.py

```diff
@@ -53,9 +56,12 @@ def create_if_has_perm(
             raise PermissionDenied
 
         created_by = principal if not isinstance(principal, PipelineRunUser) else None
+        # Before adding a random suffix, check if the couple (workspace,dataset_name) exists before
+        with_suffix = Dataset.objects.filter(workspace=workspace, slug=name).exists()
```

Comment:
This should be in the `create_dataset_slug` function

Reply:
Is it really with the name of the dataset that you want to check if there is a dataset that already exists ?

Reply:
Oh, you're right it will not work.